src/features/device-state/routes/state-patch-v3.ts

```diff
@@ -0,0 +1,365 @@
+import type { RequestHandler } from 'express';
+
+import * as _ from 'lodash';
+import {
+	captureException,
+	handleHttpErrors,
+} from '../../../infra/error-handling';
+import { sbvrUtils, errors } from '@balena/pinejs';
+import { getIP } from '../../../lib/utils';
+import {
+	Application,
+	Device,
+	Image,
+	ImageInstall,
+	PickDeferred,
+	Release,
+} from '../../../balena-model';
+import { deleteOldImageInstalls, upsertImageInstall } from './state-patch-v2';
+
+const { BadRequestError, UnauthorizedError, InternalRequestError } = errors;
+const { api } = sbvrUtils;
+
+const validPatchFields = [
+	'status',
+	'os_version',
+	'os_variant',
+	'supervisor_version',
+	'provisioning_progress',
+	'provisioning_state',
+	'ip_address',
+	'mac_address',
+	'api_port',
+	'api_secret',
+	'logs_channel',
+	'memory_usage',
+	'memory_total',
+	'storage_block_device',
+	'storage_usage',
+	'storage_total',
+	'cpu_temp',
+	'cpu_usage',
+	'cpu_id',
+	'is_undervolted',
+	'is_online',
+] as const;
+
+/**
+ * These typings should be used as a guide to what should be sent, but cannot be trusted as what actually *is* sent.
+ */
+export type StatePatchV3Body = {
+	[uuid: string]: {
+		name?: string;
+		status?: string;
```

Comment:
What is this status?

Reply:
This is the overall device status and can contain things like idle/downloading/ordered/preparing/shipping/shipped(HoD)

Reply:
this doesn't exist right now on the supervisor, is it used by the API anywhere?